### ISSUE TITLE
Che: Code completion not working with vert.x project from Burr's demo

### ISSUE THREAD

**Issue by rhopp** [openshiftio/openshift.io#525]
Code completion is not working for me with Vert.x project from Burr's demo: https://youtu.be/qyO5B2f6zHc 

I have few instances of ```[STDERR] chown: changing ownership of ‘/projects’: Operation not permitted``` when starting workspace (not sure if it is relevant).

Every time I try to invoke code completion (Ctrl+space or via top menu) I have this output in browser console:
```
VM867 _app-0.js:2029 Wed Aug 09 16:46:20 GMT+200 2017 com.google.gwt.logging.client.LogConfiguration
SEVERE: Exception caught: Fully qualified name can not be resolved for '/myvertxproject/src/main/java/io/openshift/booster/HttpApplication.java'
com.google.web.bindery.event.shared.UmbrellaException: Exception caught: Fully qualified name can not be resolved for '/myvertxproject/src/main/java/io/openshift/booster/HttpApplication.java'
	at Unknown.On(Throwable.java:74)
	at Unknown.Wn(Exception.java:33)
	at Unknown.$n(RuntimeException.java:33)
	at Unknown.gA(UmbrellaException.java:65)
	at Unknown.Qz(SimpleEventBus.java:203)
	at Unknown.HFg(SimpleEventBus.java:88)
	at Unknown.dMe(CompleteAction.java:46)
	at Unknown.WEj(PopupMenu.java:488)
	at Unknown.jFj(PopupMenu.java:637)
	at Unknown.Wnd(DOM.java:1481)
	at Unknown.zpd(DOMImplStandard.java:317)
	at Unknown.Po(Impl.java:239)
	at Unknown.So(Impl.java:291)
	at Unknown.anonymous(Impl.java:77)
	at Unknown.Fz(EventBus.java:69)
	at Unknown.rxj(SimpleEventBus.java:57)
	at Unknown.Lxj(NodeLoader.java:328)
	at Unknown.Nxj(NodeLoader.java:391)
	at Unknown.anonymous(JsPromise.java:116)
	at Unknown.Fz(EventBus.java:69)
	at Unknown.rxj(SimpleEventBus.java:57)
	at Unknown.Lxj(NodeLoader.java:328)
	at Unknown.Nxj(NodeLoader.java:391)
	at Unknown.anonymous(JsPromise.java:116)
	at Unknown.Fz(EventBus.java:69)
	at Unknown.rxj(SimpleEventBus.java:57)
	at Unknown.Lxj(NodeLoader.java:328)
	at Unknown.Nxj(NodeLoader.java:391)
	at Unknown.anonymous(JsPromise.java:116)
	at Unknown.Fz(EventBus.java:69)
	at Unknown.rxj(SimpleEventBus.java:57)
	at Unknown.sxj(NodeLoader.java:328)
	at Unknown.Lxj(NodeLoader.java:399)
	at Unknown.Nxj(NodeLoader.java:391)
	at Unknown.anonymous(JsPromise.java:116)
	at Unknown.Fz(EventBus.java:69)
	at Unknown.rxj(SimpleEventBus.java:57)
	at Unknown.sxj(NodeLoader.java:328)
	at Unknown.Lxj(NodeLoader.java:399)
	at Unknown.Nxj(NodeLoader.java:391)
	at Unknown.anonymous(JsPromise.java:116)
	at Unknown.Fz(EventBus.java:69)
	at Unknown.rxj(SimpleEventBus.java:57)
	at Unknown.sxj(NodeLoader.java:328)
	at Unknown.Lxj(NodeLoader.java:399)
	at Unknown.Nxj(NodeLoader.java:391)
	at Unknown.anonymous(JsPromise.java:116)
Caused by: java.lang.IllegalStateException: Fully qualified name can not be resolved for '/myvertxproject/src/main/java/io/openshift/booster/HttpApplication.java'
	at Unknown.Nn(Throwable.java:67)
	at Unknown.Vn(Exception.java:29)
	at Unknown.Zn(RuntimeException.java:29)
	at Unknown.RKd(IllegalStateException.java:28)
	at Unknown.z5h(JavaUtil.java:140)
	at Unknown.yGh(JavaUtil.java:150)
	at Unknown.IDf(CodeAssistantImpl.java:60)
	at Unknown.vFg(OrionEditorInit.java:264)
	at Unknown.YHg(OrionEditorWidget.java:635)
	at Unknown.hGg(TextEditorPartViewImpl.java:79)
	at Unknown.kFg(OrionEditorInit.java:258)
	at Unknown.qFg(OrionEditorInit.java:217)
	at Unknown.HEf(CompletionRequestEvent.java:29)
	at Unknown.Qz(GwtEvent.java:76)
	at Unknown.HFg(SimpleEventBus.java:88)
	at Unknown.dMe(CompleteAction.java:46)
	at Unknown.WEj(PopupMenu.java:488)
	at Unknown.jFj(PopupMenu.java:637)
	at Unknown.Wnd(DOM.java:1481)
	at Unknown.zpd(DOMImplStandard.java:317)
	at Unknown.Po(Impl.java:239)
	at Unknown.So(Impl.java:291)
	at Unknown.anonymous(Impl.java:77)
	at Unknown.Fz(EventBus.java:69)
	at Unknown.rxj(SimpleEventBus.java:57)
	at Unknown.Lxj(NodeLoader.java:328)
	at Unknown.Nxj(NodeLoader.java:391)
	at Unknown.anonymous(JsPromise.java:116)
	at Unknown.Fz(EventBus.java:69)
	at Unknown.rxj(SimpleEventBus.java:57)
	at Unknown.Lxj(NodeLoader.java:328)
	at Unknown.Nxj(NodeLoader.java:391)
	at Unknown.anonymous(JsPromise.java:116)
	at Unknown.Fz(EventBus.java:69)
	at Unknown.rxj(SimpleEventBus.java:57)
	at Unknown.Lxj(NodeLoader.java:328)
	at Unknown.Nxj(NodeLoader.java:391)
	at Unknown.anonymous(JsPromise.java:116)
	at Unknown.Fz(EventBus.java:69)
	at Unknown.rxj(SimpleEventBus.java:57)
	at Unknown.sxj(NodeLoader.java:328)
	at Unknown.Lxj(NodeLoader.java:399)
	at Unknown.Nxj(NodeLoader.java:391)
	at Unknown.anonymous(JsPromise.java:116)
	at Unknown.Fz(EventBus.java:69)
	at Unknown.rxj(SimpleEventBus.java:57)
	at Unknown.sxj(NodeLoader.java:328)
	at Unknown.Lxj(NodeLoader.java:399)
	at Unknown.Nxj(NodeLoader.java:391)
	at Unknown.anonymous(JsPromise.java:116)
	at Unknown.Fz(EventBus.java:69)
	at Unknown.rxj(SimpleEventBus.java:57)
	at Unknown.sxj(NodeLoader.java:328)
	at Unknown.Lxj(NodeLoader.java:399)
	at Unknown.Nxj(NodeLoader.java:391)
	at Unknown.anonymous(JsPromise.java:116)

obd @ VM867 _app-0.js:2029
h_d @ VM867 _app-0.js:3682
g_d @ VM867 _app-0.js:3681
j_d @ VM867 _app-0.js:3684
Wo @ VM867 _app-0.js:404
So @ VM867 _app-0.js:400
(anonymous) @ VM867 _app-0.js:399
```

**Comment by rhopp** [comment]
After changing project type to "Maven" (Project -> Update project configuration... -> select Java-Maven -> Save) code completion works. I also noticed, that now package is correctly recognized and correctly shown in IDE. Before it was just lot of nested folders (io/openshift/booster/).

**Comment by slemeur** [comment]
I have the same issue as well recorded a video for it.

![project-type](https://user-images.githubusercontent.com/1636769/29463088-b8fd9c40-8431-11e7-9c3c-4718ca2f6668.gif)

**Comment by ibuziuk** [comment]
@eivantsov could this be related to https://github.com/eclipse/che/issues/5970?

**Comment by eivantsov** [comment]
Yeah, I think at this point we don't have AST. Once the problem with resolving project deps is resolved, this one disappears as well.

**Comment by ibuziuk** [comment]
Che 294 should contain fixes for https://github.com/eclipse/che/issues/5970

**Comment by ibuziuk** [comment]
@rhopp could you please verify on osio against 294 ?

**Comment by rafabene** [comment]
I've tried it today and I was never able to reproduce it. Can we close this?

**Comment by ibuziuk** [comment]
294 contains fixes for eclipse/che#5970. Closing

**Comment by joshuawilson** [comment]
@ibuziuk is 294?
